docs/tasks/debug-application-cluster/local-debugging.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Developing and debugging services locally
+---
+
+{% capture overview %}
+
+Kubernetes applications usually consist of multiple, separate services, each running in its own container. Developing and debugging these services on a remote Kubernetes cluster can be cumbersome, requiring you to [get a shell on a running container](https://kubernetes.io/docs/tasks/debug-application-cluster/get-shell-running-container/) and then running your tools inside the remote shell.
+
+This document describes using `telepresence` to develop and debug a service locally, while proxying the service to a remote Kubernetes cluster. Using `telepresence` allows you to use custom tools, such as a debugger and IDE, for a local service and provides the service full access to ConfigMap, secrets, and the services running on the remote cluster.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* Kubernetes cluster is installed
+* `kubectl` is configured to communicate with the cluster
+* [Telepresence](https://www.telepresence.io/reference/install) is installed
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Getting a shell on a remote cluster
+
+If you start `telepresence` with no arguments, you'll get a special shell. Unlike a remote shell, the telepresence shell runs locally, giving you full access to your filesystem.
+
+There are a variety of ways that the local shell can be used. For example, you could write a shell script on your laptop, and run it directly from the shell in real-time. You can do this on a remote shell as well, of course, but you may not be able to use your preferred code editor, and the script is deleted when you terminate the container.
+
+You can exit the shell by typing `exit`.
+
+## Developing or debugging an existing service
+
+When developing an application on Kubernetes, you typically code/debug a single service, but that service may require access to other services for proper testing and debugging. One option for coding and debugging is to use your continuous deployment pipeline, but even the fastest deployment pipeline introduces a delay in the code/debug cycle.
+
+With the `--swap-deployment` option, Telepresence lets you swap an existing deployment with the Telepresence proxy. This lets you run your service locally while connecting to the remote Kubernetes cluster -- while letting other services in the remote cluster also access your locally running instance. To run Telepresence with `--swap-deployment`, type:
```

Comment:
Use the `--swap-deployment` option to swap an existing deployment with the Telepresence proxy. Swapping allows you to run a service locally and connect to the remote Kubernetes cluster. The services in the remote cluster can now access the locally running instance. 

To run `telepresence` with `--swap-deployment`, enter: